From 4a04cfd6a381c71577d8dbcfa614d550499b1226 Mon Sep 17 00:00:00 2001
From: Grygorii Strashko <grygorii.strashko@ti.com>
Date: Mon, 28 Dec 2015 20:00:11 +0200
Subject: [PATCH 3/3] kernel/time/timer.c: UP: fix undefined reference to
 `del_timer_sync'

This patch fixes below build errors in case SMP=n:

drivers/built-in.o: In function `fbcon_del_cursor_timer':
/home/x0174654/kernel.org/linux-master/linux/drivers/video/console/fbcon.c:433: undefined reference to `del_timer_sync'
drivers/built-in.o: In function `sysrq_disconnect':
/home/x0174654/kernel.org/linux-master/linux/drivers/tty/sysrq.c:928: undefined reference to `del_timer_sync'
drivers/built-in.o: In function `kd_mksound':
/home/x0174654/kernel.org/linux-master/linux/drivers/tty/vt/keyboard.c:256: undefined reference to `del_timer_sync'
drivers/built-in.o: In function `do_blank_screen':
/home/x0174654/kernel.org/linux-master/linux/drivers/tty/vt/vt.c:3842: undefined reference to `del_timer_sync'
drivers/built-in.o: In function `univ8250_release_irq':
/home/x0174654/kernel.org/linux-master/linux/drivers/tty/serial/8250/8250_core.c:365: undefined reference to `del_timer_sync'
drivers/built-in.o:/home/x0174654/kernel.org/linux-master/linux/drivers/base/power/wakeup.c:114: more undefined references to `del_timer_sync' follow

Signed-off-by: Grygorii Strashko <grygorii.strashko@ti.com>
---
 kernel/time/timer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/time/timer.c b/kernel/time/timer.c
index c3314fc..a460cbb 100644
--- a/kernel/time/timer.c
+++ b/kernel/time/timer.c
@@ -1118,7 +1118,7 @@ int try_to_del_timer_sync(struct timer_list *timer)
 }
 EXPORT_SYMBOL(try_to_del_timer_sync);
 
-#ifdef CONFIG_SMP
+#if defined(CONFIG_SMP) || defined(CONFIG_PREEMPT_RT_FULL)
 /**
  * del_timer_sync - deactivate a timer and wait for the handler to finish.
  * @timer: the timer to be deactivated
-- 
2.6.4

